dashboard/database.py

```diff
@@ -0,0 +1,135 @@
+"""Functions for interacting with the database."""
+
+from os import environ as ENV
+
+from psycopg import connect, Connection
+from dotenv import load_dotenv
+import streamlit as st
+import pandas as pd
+
+
+@st.cache_resource
+def get_connection() -> Connection:
+    """gets a connection"""
+    connection = connect(
+        port=ENV["DB_PORT"],
+        dbname=ENV["DB_NAME"],
+        host=ENV["DB_ENDPOINT"],
+        user=ENV["DB_USER"],
+        password=ENV["DB_PASSWORD"]
+    )
+    return connection
+
+
+@st.cache_data(ttl="1hr")
+def get_popular_tracks(_conn: Connection, n: int = 5) -> pd.DataFrame:
+    """Returns the N most sold tracks in the database."""
+
+    print("Collating most popular tracks.")
```

Comment:
Not a major issue but I think an ellipses is better here, and anywhere similar.
```suggestion
    print("Collating most popular tracks...")
```